app/plugins/adp-backend/src/service/deliveryProgrammeAdminRouter.ts

```diff
@@ -0,0 +1,171 @@
+import { errorHandler } from '@backstage/backend-common';
+import { DiscoveryService } from '@backstage/backend-plugin-api';
+import { IdentityApi } from '@backstage/plugin-auth-node';
+import { Logger } from 'winston';
+import { IDeliveryProgrammeAdminStore } from '../deliveryProgrammeAdmin';
+import express from 'express';
+import Router from 'express-promise-router';
+import { InputError } from '@backstage/errors';
+import { CatalogClient } from '@backstage/catalog-client';
+import { UserEntityV1alpha1 } from '@backstage/catalog-model';
+import { CreateDeliveryProgrammeAdmin } from '../utils';
+
+type CreateDeliveryProgrammeAdminRequest = {
+  aadEntityRefId: string;
+};
+
+type DeleteDeliveryProgrammeAdminRequest = {
+  aadEntityRefId: string;
+  deliveryProgrammeId: string;
+};
+
+export interface DeliveryProgrammeAdminRouterOptions {
+  logger: Logger;
+  identity: IdentityApi;
+  deliveryProgrammeAdminStore: IDeliveryProgrammeAdminStore
+  discovery: DiscoveryService;
+}
+
+export async function createDeliveryProgrammeAdminRouter(
+  options: DeliveryProgrammeAdminRouterOptions,
+): Promise<express.Router> {
+  const { logger, discovery, deliveryProgrammeAdminStore } = options;
+  const catalog = new CatalogClient({ discoveryApi: discovery });
```

Comment:
This could be injected in through the options the same way the deliveryProgrammeAdminStore is. That way you wouldnt need to mock the catalogclient module in the test